File: src/main/java/com/dozycoffee/wms/common_code/domain/model/CommonCode.java
package com.dozycoffee.wms.common_code.domain.model;

import lombok.Getter;

@Getter
public class CommonCode {

    private final String code;
    private final String groupCode;
    private final String name;
    private final int sortOrder;
    private final boolean active;

    private CommonCode(String code, String groupCode, String name, int sortOrder, boolean active) {
        this.code = code;
        this.groupCode = groupCode;
        this.name = name;
        this.sortOrder = sortOrder;
        this.active = active;
    }

    public static CommonCode create(String code, String groupCode, String name, int sortOrder) {
        return new CommonCode(code, groupCode, name, sortOrder, true);
    }

    public static CommonCode reconstruct(String code, String groupCode, String name, int sortOrder, boolean active) {
        return new CommonCode(code, groupCode, name, sortOrder, active);
    }
}

File: src/main/kotlin/com/dozycoffee/wms/common_code/domain/model/CommonCode.kt
package com.dozycoffee.wms.common_code.domain.model

class CommonCode private constructor(
    val code: String,
    val groupCode: String,
    val name: String,
    val sortOrder: Int,
    val active: Boolean
) {
    companion object {
        fun create(code: String, groupCode: String, name: String, sortOrder: Int): CommonCode {
            return CommonCode(code, groupCode, name, sortOrder, true)
        }

        fun reconstruct(code: String, groupCode: String, name: String, sortOrder: Int, active: Boolean): CommonCode {
            return CommonCode(code, groupCode, name, sortOrder, active)
        }
    }
}
